### DISCUSSION TITLE
[Documentation]: Storybook backend not found when using static build but works in development

### DISCUSSION THREAD

**Discussion by Ezmr7** [storybookjs/storybook#31506] [Help]
### Describe the problem

Hello. I'm working on an addon that accesses Storybook's backend through a middleware.js file to make requests to external sources. This works perfectly for development, but upon building the static build, any requests made in the frontend give a 404 error and aren't ever reaching the backend. I'm wondering if there is an alternative undocumented method to make requests that I'm unaware of. As the Storybook Designs addon seems to make requests without a middleware.js file. Any help is much appreciated! Thanks!

**Comment by valentinpalkovic** [comment]
As you have figured out, Storybook doesn't have any backend as soon as it is built. If you want to do any requests, you have to request them from the preview, instead of the `middleware.js`, which is only executed when a backend exists (in dev mode)
